### DISCUSSION TITLE
Signup using Google

### DISCUSSION THREAD

**Discussion by renieness** [SwipesForScience/SwipesForScience#120] [General]
I feel like we've spoken about this in the past, but would we ever want to allow Players to create accounts using Google auth or something else?

This would probably differ between community science games vs educational games (for lack of a better phrase).

**Comment by richford** [comment]
Yeah, I like this idea. Currently we allow only email/password sign-in but we could allow a lot of different types:

![image](https://user-images.githubusercontent.com/7978135/111627765-a78b9700-87ac-11eb-9497-a686f12bda2e.png)

Ideally, the scientific user would specify the allowed types in their config file and then the app would include those types accordingly. This is a long list but I think we could start with Google and GitHub.

For google, we could follow these instructions: https://vue-view.com/making-a-one-click-google-sign-in-for-vue-js-application/

**Reply by jyeatman** [reply]
I agree! Flexibility in sign ups for different applications would be really nice. How does anonymous work? That might be useful in some scenarios too

**Reply by renieness** [reply]
Yeah the anonymous thing would've made more sense in the community game scene. 

Part of the reason for it was to entice more players into playing a game. I thought that if you were presented with a sign-up screen straightaway, you may be put off playing. 

Conversely though, in the Appstract game example, Anisha mentioned that she wanted to almost grade the player playing - to figure out how reliable an annotator they were. 

From this angle, and the fact that anonymity makes the code a bit more complex, I think we could probably remove it for now, and add it back in later, if we need it.

What scenarios could anonymity be useful in, @jyeatman?

Also, wow, there are a lot of types of authentication we could use!!!

**Reply by jyeatman** [reply]
I agree. I'm don't fully understand how the anonymous sign in works. But if we created a sign in option with randomly generated usernames then it is essentially anonymous but also tracks the subject over time.
